### PR TITLE
Publisher confirms

### PR DESCRIPTION
**Pull request by mtowers**

### CODE REVIEW

File: lib/realself/stream/publisher.rb [hunk 1]
```diff
@@ -103,28 +95,19 @@ def publish(item, routing_key, content_type = 'application/json')
...
 
 
-      def initialize_connection
-        # clean up any open channel
-        close_channel
-
+      def ensure_connection!
         # start a new connection
-        @publisher_session  = Bunny.new(symbolize_keys(@rmq_config))
+        @publisher_session  ||= Bunny.new(symbolize_keys(@rmq_config))
```
Comment (wulftone) [inline lib/realself/stream/publisher.rb]: nice

**Comment by wulftone** [comment]
looks good to me, did you put it up in testvpc?

**Comment by mtowers** [comment]
@wulftone We did extensive testing in TESTVPC with this build.

**Comment by wulftone** [comment]
:rainbow:

**Pull request merged**